### DISCUSSION TITLE
Forcing angle in crystalline structure

### DISCUSSION THREAD

**Discussion by fabiospampinato** [d3/d3#3509] [Q&A]
Hello there, I'm using d3-force for a little simulation where nodes have labels attached to them visually.

The problem is that if nodes happen to align ~horizontally then those labels are really crammed in there visually:

<img width="849" alt="Screen Shot 2021-06-16 at 15 40 16" src="https://user-images.githubusercontent.com/1812093/122241294-59c12080-ceba-11eb-9357-69f59290d43a.png">

While if nodes are aligned ~vertically labels are laid out much better:

<img width="842" alt="Screen Shot 2021-06-16 at 15 40 27" src="https://user-images.githubusercontent.com/1812093/122241489-7f4e2a00-ceba-11eb-9964-0d453a1c27fd.png">

Those two crystalline (?) structures are both hexagonal, but basically the difference is that if the pointy tips of the hexagon are aligned vertically the thing looks great, while if they are aligned horizontally the thing looks pretty bad.

What would be the best way to force the pointy tips to align vertically? I'm thinking that maybe I should change the collision force, since the default one expects point-like masses, should that work? If so is there some already-done alternative collision force that I could use for this?

**Comment by fabiospampinato** [comment]
There's [d3-bboxCollide](https://github.com/emeeks/d3-bboxCollide) already available which in theory could partially address the problem, but I couldn't get it to work at all, possibly the only option is to patch the built-in collision force for this, hopefully detecting overlapping hexagons isn't too much of a pain in the ass to code.

**Reply by Fil** [reply]
You can find several [examples of d3-bboxCollide on Observable](https://observablehq.com/search?query=d3-bboxCollide). Although it hasn't been updated for years, it still works with d3@7:
https://observablehq.com/@fil/hello-d3-bboxcollide
